### PR TITLE
Anilist v2 (#343)

### LINKED ISSUE #343: AniList API v1 shutdown - Migration to v2
AniList API v1 was given 6 months before it was meant to be shut down after API v2 went stable.
I made a few announcements about this and sent out emails to developers who had registered API v1 clients.

It looks like trackma is still using API v1, which was meant to be shut down on May 1st, due to some issues on AniList's part, API v1 is still running, though will likely only up for another 2 weeks or so, as the majority of applications have already migrated to v2.

Super sorry if you didn't hear about all this, as this would be super short notice.

[Api v2 docs can be found here](https://github.com/AniList/ApiV2-GraphQL-Docs), if you check the migration folder theres a number of queries that aim to replicate v1's rest call data set, so hopefully they help with the possible migration.

### PR DESCRIPTION
**Pull request by Birdulon**
Had a bit more time to finish off my implementation. Feel free to close it if you've made significant progress with yours, I needed to learn the new API for my own reasons anyway.
Also if you decide to merge, feel free to change the client ID and secret pair to something under your account. (fixes #343)

### CODE REVIEW

**Comment by z411** [comment]
Thank you. I had already modified the basic functions but hadn't implemented the calls yet so I will review this and merge it with what I had done.

Also a concern of mine is the token; wouldn't the token refresh fail since the authorization code expires after its use?

**Comment by z411** [comment]
Thank you. I had already modified the basic functions but hadn't implemented the calls yet so I will review this and merge it with what I had done.

Also a concern of mine is the token; wouldn't the token refresh fail since the authorization code expires after its use?

File: trackma/lib/libanilist.py [hunk 1]
```diff
@@ -70,13 +70,13 @@ class libanilist(lib):
...
-            'reading': 'Reading',
-            'completed': 'Completed',
-            'on-hold': 'On Hold',
-            'dropped': 'Dropped',
-            'plan to read': 'Plan to Read'
+            'CURRENT': 'Watching',
+            'COMPLETED': 'Completed',
+            'PAUSED': 'On Hold',
+            'DROPPED': 'Dropped',
+            'PLANNING': 'Plan to Ratch'
```
Comment (FichteFoll) [inline trackma/lib/libanilist.py]: `Watch` typo here.

Comment (Birdulon) [reply trackma/lib/libanilist.py]: Good catch!

**Comment by Birdulon** [comment]
Re: Token - I'm not entirely sure what's going on with it, but my user.json does have different access_token and refresh_tokens. Token expiration is a year away so unsure how to otherwise test that.

**Comment by z411** [comment]
Merged most of your changes into the `anilistv2` branch, thank you.

**Pull request closed without merging**